src/frontend/src/flows/manage.ts

```diff
@@ -150,94 +203,82 @@ export const renderManage = async (
   let devices: DeviceData[];
   try {
     devices = await withLoader(() => IIConnection.lookupAll(userNumber));
-  } catch (err) {
-    await displayError({
-      title: "Failed to list your devices",
-      message:
-        "An unexpected error occurred when displaying your devices. Please try again",
-      detail: err.toString(),
-      primaryButton: "Try again",
-    });
+  } catch (error) {
+    await displayFailedToListDevices(error);
     return renderManage(userNumber, connection);
   }
   render(pageContent(userNumber, devices), container);
   init(userNumber, connection, devices);
 };
 
+// Add a new device (i.e. a device connected to the device the user is
+// currently using, like a YubiKey, or FaceID, or, or. Not meant to be used to
+// add e.g. _another_ macbook or iPhone.)
+const addAdditionalDevice = async (
+  userNumber: bigint,
+  connection: IIConnection,
+  devices: DeviceData[]
+) => {
+  let newDevice: WebAuthnIdentity;
+  try {
+    newDevice = await WebAuthnIdentity.create({
+      publicKey: creationOptions(devices),
+    });
+  } catch (error) {
+    await displayFailedToAddNewDevice(error);
+    return renderManage(userNumber, connection);
+  }
+  const deviceName = await pickDeviceAlias();
+  if (deviceName === null) {
+    return renderManage(userNumber, connection);
+  }
+  // TODO check whether newDevice is already registered,
+  // or better, pass existing devices to `.create` so that they
+  // cannot be added again
```

Comment:
We do pass the existing devices to `.create` in line 225, so I don't think there's anything we can do here?

Reply:
I didn't write that :)

Reply:
In that case, even better!

```suggestion
```